backend/src/controllers/AuthController.ts

```diff
@@ -108,3 +108,20 @@ export const register = async (req: AuthRequest, res: Response) => {
     });
   }
 };
+
+export const checkIfTeacher = async (req: AuthRequest, res: Response, next: NextFunction) => {
+  const user = req.user
+  if (!user){
+    return res.status(400).json({
+      status: "error",
+      msg: "This user does not have in database",
+    });
+  }
+  if (user!.roles != "teacher") {
```

Comment:
```suggestion
  if (user!.role !== "teacher") {
```
You have a typo here, thus causing the value to be `undefined` which cause the request to stop here in `/api/v1/tests`